Sources/Apodini/Configurations/HTTPConfiguration.swift

```diff
@@ -8,75 +8,96 @@
-            case .incompatibleFlags:
-                return "The command line arguments for HTTPConfiguration are invalid."
-            }
-        }
-
-        var recoverySuggestion: String? {
-            switch self {
-            case .incompatibleFlags:
-                return "Example usage of HTTPConfiguration: --hostname 0.0.0.0 --port 8080 or --bind 0.0.0.0:8080"
+    
+    public var bindAddress: BindAddress
+    public var hostname: Hostname
+    public var supportVersions: Set<HTTPVersionMajor> = [.one]
+    public var tlsConfiguration: TLSConfiguration?
+    
+    
+    public var uriPrefix: String {
+            let httpProtocol: String
+            var port = ""
+            
+            if self.tlsConfiguration == nil {
+                httpProtocol = "http://"
+                if hostname.port != 80 {
+                    port = ":\(hostname.port)"
+                }
+            } else {
+                httpProtocol = "https://"
+                if hostname.port != 443 {
+                    port = ":\(hostname.port)"
+                }
             }
-        }
+            
+        return httpProtocol + hostname.address + port
     }
     
-    private var address: BindAddress?
     
     /// initalize HTTPConfiguration
-    public init(hostname: String? = nil, port: Int? = nil, bind: String? = nil, socketPath: String? = nil) {
-        do {
-            switch (hostname, port, bind, socketPath) {
-            case (.none, .none, .none, .none):
-                self.address = nil
-            case (.none, .none, .none, .some(let socketPath)):
-                self.address = .unixDomainSocket(path: socketPath)
-            case (.none, .none, .some(let address), .none):
-                let components = address.split(separator: ":")
-                let hostname = components.first.map { String($0) }
-                let port = components.last.flatMap { Int($0) }
-                self.address = .hostname(hostname, port: port)
-            case let (hostname, port, .none, .none):
-                self.address = .hostname(hostname ?? Defaults.hostname, port: port ?? Defaults.port)
-            default:
-                throw HTTPConfigurationError.incompatibleFlags
+    public init(hostname: Hostname? = nil, bindAddress: BindAddress? = nil, tlsConfigurationBuilder: TLSConfigurationBuilder? = nil) {
+        var defaultPort = Defaults.httpPort
+        
+        if let tlsConfigBuilder = tlsConfigurationBuilder {
+            self.supportVersions.insert(.two)
+            self.tlsConfiguration = tlsConfigBuilder.tlsConfiguration
+            defaultPort = Defaults.httpsPort
+        }
+        
+        if let bindAddress = bindAddress {
+            switch bindAddress {
+            case let .interface(address, port):
+                if port != nil {
+                    self.bindAddress = bindAddress
+                } else {
+                    self.bindAddress = .interface(address, port: defaultPort)
+                }
```

Comment:
If you want you can take advantage of the power of Swift switch statements here: 
```suggestion
            switch bindAddress {
            case let .interface(address, nil):
                self.bindAddress = .interface(address, port: defaultPort)
            case .interface:
                self.bindAddress = bindAddress
```

Same apples to line 82 and following below